wire-ios/Wire-iOS/Sources/UserInterface/Conversation/ConversationViewController+ProfileViewControllerDelegate.swift

```diff
@@ -31,15 +31,17 @@ extension ConversationViewController {
             profileViewControllerDelegate: self,
             viewControllerDismisser: self,
             userSession: userSession,
-            mainCoordinator: mainCoordinator
+            mainCoordinator: mainCoordinator,
+            selfProfileUIBuilder: selfProfileUIBuilder
         )
     }
 }
 
 extension ConversationViewController: ProfileViewControllerDelegate {
     func profileViewController(_ controller: ProfileViewController?, wantsToNavigateTo conversation: ZMConversation) {
-        dismiss(animated: true) {
-            self.mainCoordinator.openConversation(conversation, focusOnView: true, animated: true)
+        Task {
+            await mainCoordinator.showConversationList(conversationFilter: .none)
+            await mainCoordinator.showConversation(conversation: conversation, message: nil)
```

Comment:
why do we showConversationList before? also does this remove the current conversationFilter?

Reply:
We only want to show a conversation in the secondary column when the supplementary (middle) column shows the conversation list.
Do you prefer to implicitly show the conversation list?

About the filter: yes, this will remove the current filter. 
If we want to prevent it, we need a more complex logic here. We would need to check which is the current filter and if the conversation is or will be visible with this filter. 

Thinks about the following examples:
1. We filter by group conversations, open a profile and want to open the 1:1 conversation of this user. Shall we switch to 1:1 or to all conversations?
2. What if the conversation exists and is marked as favorite? Do we then want to jump to the favorites?
3. What if we had not filtered the conversation list before? We probably want to stay with the same filter.

I suggest to keep it this way and observe if the behavior can be improved.

Reply:
ok agreed, maybe put this into a follow up ticket that we can refine on the desired behaviour